src/gui/EditWidgetIcons.cpp

```diff
@@ -273,50 +273,77 @@ void EditWidgetIcons::removeCustomIcon()
         QModelIndex index = m_ui->customIconsView->currentIndex();
         if (index.isValid()) {
             Uuid iconUuid = m_customIconModel->uuidFromIndex(index);
-            int iconUsedCount = 0;
 
             const QList<Entry*> allEntries = m_database->rootGroup()->entriesRecursive(true);
+            QList<Entry*> entriesWithSameIcon;
             QList<Entry*> historyEntriesWithSameIcon;
 
             for (Entry* entry : allEntries) {
-                bool isHistoryEntry = !entry->group();
                 if (iconUuid == entry->iconUuid()) {
-                    if (isHistoryEntry) {
+                    // Check if this is a history entry (no assigned group)
+                    if (entry->group() == nullptr) {
                         historyEntriesWithSameIcon << entry;
                     }
-                    else if (m_currentUuid != entry->uuid()) {
-                        iconUsedCount++;
+                    else if (m_currentUuid != entry->uuid()){
```

Comment:
You should reformat the else if statements, so we can slowly transition to the Qt coding standard and away from this ugly newline before else.